src/extension/features/accounts/bulk-edit-memo/index.jsx

```diff
@@ -5,57 +5,145 @@ import { l10n } from 'toolkit/extension/utils/toolkit';
 import { componentAfter } from 'toolkit/extension/utils/react';
 import { getEntityManager } from 'toolkit/extension/utils/ynab';
 
+const DEFAULT_DISPLAY_MODE = 'defaultDisplayMode';
+const MENU_DISPLAY_MODE = 'menuDisplayMode';
+const EDIT_DISPLAY_MODE = 'editDisplayMode';
+
+const REPLACE_EDIT_MODE = 'replaceEditMode';
+const ADD_PREFIX_EDIT_MODE = 'addPrefixEditMode';
+const ADD_SUFFIX_EDIT_MODE = 'addSuffixEditMode';
+
 const EditMemo = () => {
-  const [isEditMode, setIsEditMode] = useState(false);
+  const [displayMode, setDisplayMode] = useState(DEFAULT_DISPLAY_MODE);
+  const [editMode, setEditMode] = useState(REPLACE_EDIT_MODE);
   const [memoInputValue, setMemoInputValue] = useState('');
 
+  const makeNewMemo = (prevMemo) => {
+    switch (editMode) {
+      case REPLACE_EDIT_MODE:
+        return memoInputValue;
+      case ADD_PREFIX_EDIT_MODE:
+        return memoInputValue + prevMemo;
+      case ADD_SUFFIX_EDIT_MODE:
+        return prevMemo + memoInputValue;
```

Comment:
I'm thinking we should maybe add a space here for the user? it might be confusing if their prefix just gets added without the space?

Reply:
I was considering this as well.

If we don't insert a whitespace, a user can press Ctrl+z to cancel an undesired update (works with updating several transactions, I checked), then they can repeat the operation, but add a whitespace.

If we force inserting a whitespace, some user might be unhappy if they want to add a prefix/suffix without one. However, this seems very unlikely and they can raise an issue in the tracker.

The way to make everyone happy would be to add a checkbox saying "add a whitespace" that's enabled by default. Might look a bit too overloaded though.

WDYT? I personally prefer the 3rd way, but don't insist.